### ISSUE TITLE
Is there any problem with this training result?

### ISSUE THREAD

**Issue by rivergold** [JiahuiYu/generative_inpainting#434]
I rewrite train code into PyTorch, and test train result which is shown as follows.
**Mask**
![image](https://user-images.githubusercontent.com/6269549/82981976-c81d1180-a01f-11ea-94c4-30f148b0fbc2.png)
**Train data result**
![image](https://user-images.githubusercontent.com/6269549/82981949-b6d40500-a01f-11ea-8d8b-2b7a6a1f3e03.png)
**Loss curve**
![image](https://user-images.githubusercontent.com/6269549/82982018-dc610e80-a01f-11ea-92e3-ac9224b63528.png)
I'm not sure if this result is correct and should continue or not…

**Comment by JiahuiYu** [comment]
@rivergold It's brave to rewrite the code in pytorch which I thought for a while..

So it seems the GAN loss may have issue. Are you working on a large dataset? What's the training batch size?

Be very careful if you want to modify the training hyper-parameters. Best of luck!

**Comment by rivergold** [comment]
@JiahuiYu Thanks for the replay and your code is really great!
Because I'm more familiar with PyTorch and want to have a try. And rewriting can help me learn more about details.
I use the whole places365 dataset and batch_size=24. All hyper-parameters are the same as raw code.
After about 40k step, the loss is like this:
![image](https://user-images.githubusercontent.com/6269549/83107293-93c55600-a0f0-11ea-9e06-a1f13bbc11aa.png)
- g_loss: hinge loss for negative samples
- g_loss_all: g_loss + l1_loss

**Mask**
![image](https://user-images.githubusercontent.com/6269549/83107446-e3a41d00-a0f0-11ea-900b-d94202edb078.png)
**Inpainted**
![image](https://user-images.githubusercontent.com/6269549/83107531-0a625380-a0f1-11ea-9045-e48c146bebd5.png)
Some results are good, but I'm confused with the stage 2 output, it will occur some strange part during training which is shown as followings. Is it correct?
![image](https://user-images.githubusercontent.com/6269549/83107914-9bd1c580-a0f1-11ea-878a-c6cdaf437ebc.png)
![image](https://user-images.githubusercontent.com/6269549/83108032-c885dd00-a0f1-11ea-9096-99acab12bc9f.png)

**Comment by JiahuiYu** [comment]
@rivergold In my view, the bad results happen in cases when the l1/l2 loss balancing weight is not large enough, thus GAN loss dominates the gradients. The artifacts/patterns are very likely from the failure of GANs.

Typically if you train the model for more than 48 hours, the results should be good enough. Otherwise you might need to retrain the model by tuning some of the hyper-parameters.
